### ISSUE TITLE
Thorntail-runner version 2.4.0.Final JAX-RS application class not found

### ISSUE THREAD

**Issue by bojanb1234** [thorntail/thorntail#1276]
Reason for a failure is inside a WarBuilder class inside a method isWar. Paths should be normalized:

private boolean isWar(String path) {
        String currentDir = Paths.get(".").toAbsolutePath()**.normalize()**.toString();
        String classesDirPath = Paths.get(path).toAbsolutePath()**.normalize()**.toString();
        return classesDirPath.startsWith(currentDir);
    }

If paths are not normalized, than a WarBuilder moves files from classes directory in a new jar which is than packed inside lib folder. During application deployment an inherited class from javax.ws.rs.core.Application cannot be loaded from inside an new jar.

**Comment by Ladicek** [comment]
This is about Thorntail 2 and hence belongs to JIRA: https://issues.jboss.org/browse/THORN Also CC @michalszynkiewicz

**Comment by bojanb1234** [comment]
Opened a Jira isue:  https://issues.jboss.org/browse/THORN-2393

**Comment by bojanb1234** [comment]
File path normalization resolved ClassNotFoundException with JAX-RS Application class only on single module project. In multi-module project the problem remains.

Thorntail-runner dynamically creates users.war which includes in WEB-INF\lib folder dynamically created jars (thorntail-user-jar-....) from dependent module class paths (/target/classes folders from dependent modules). Classes from this thorntail-user-jars-... cannot be loaded. I have inspected the content of those jars and they seams to be ok.

I have try to solve the problem from other side. I debug ModuleClassLoader when running application. I have discovered, that org.jboss.as.server.deployment.module.VFSResourceLoader.getPaths() methods return empty array for thorntail-user-jars-... Reason for that is this part of code, which seeks for directories inside jar. 

FilterVirtualFileVisitor visitor = new FilterVirtualFileVisitor(new VirtualFileFilter() {
            @Override
            public boolean accepts(VirtualFile file) {
                return file.isDirectory();
            }
        }, VisitorAttributes.RECURSE);

Looking inside thorntail-user-jar-... reveles that jar contains folders. Looking at the code i noticed two things:
- Zip should contains forward slash for file separator regardles of operation system. I used URI for handle with this problem
- Adding files to zip without folders. This is ok for unziping, but VFSResourceLoader expects folders. I change the logic for ziping those jars.
